source/common/ur_pool_manager.hpp

```diff
@@ -191,6 +206,357 @@ pool_descriptor::create(ur_usm_pool_handle_t poolHandle,
     return {ret, descriptors};
 }
 
+struct pool_config {
+    // TODO: Config influenced by env variables
+    // dimensions[memType] = {allocMax(MB), capacity, poolSize(MB)}
+    std::map<int32_t, std::vector<int32_t>> dimensions = {
+        {UR_USM_TYPE_DEVICE, {1, 4, 256}},
+        {UR_USM_TYPE_HOST, {1, 4, 256}},
+        {UR_USM_TYPE_SHARED, {8, 4, 256}}};
+} PoolConfig;
+
+struct proxy_pool {
```

Comment:
What is the responsibility of this struct?

Reply:
It was meant to forward the alloc/dealloc calls to the memory provider and perform pooling.

Reply:
So, is it an alternative to the disjoint pool? Is it just one more implementation of the heap manager?

Reply:
I think this implementation is no longer needed (at least for now), the PR was not just updated yet.

Reply:
Got it. It was an intermediate solution while we deal with license issues.